test/lit/passes/tuple-optimization.wast

```diff
@@ -580,12 +580,12 @@
   ;; CHECK-NEXT:   (unreachable)
   ;; CHECK-NEXT:  )
   ;; CHECK-NEXT:  (drop
-  ;; CHECK-NEXT:   (tuple.extract 1 0
+  ;; CHECK-NEXT:   (tuple.extract 2 0
   ;; CHECK-NEXT:    (unreachable)
   ;; CHECK-NEXT:   )
   ;; CHECK-NEXT:  )
   ;; CHECK-NEXT:  (drop
-  ;; CHECK-NEXT:   (tuple.extract 1 1
+  ;; CHECK-NEXT:   (tuple.extract 2 1
   ;; CHECK-NEXT:    (local.tee $tuple
```

Comment:
Doesn't this depend on the type of `$tuple`, which happens to be of size 2? That is, if line 571 changed to have 17 items in the tuple, would this still validate?

Reply:
Yes, it would still validate because we do not store the expected arity of `TupleExtract` in the IR, so the validator would not know what the expected arity is, just that the tuple value is unreachable.